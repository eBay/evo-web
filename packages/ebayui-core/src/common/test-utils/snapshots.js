import { render, prettyDOM } from "@marko/testing-library";
import { expect } from "vitest";
import visualHTML from "visual-html";
import { diffLines } from "diff";

function snapshotHTML() {
    return async (template, input) => {
        const { container } = await render(template, input);
        expect(prettyDOM(container)).toMatchSnapshot();
    };
}

async function diffServerSnapshots(template, input) {
    const { container } = await render(template, input);
    return async (afterTemplate, afterInput) => {
        const { container: afterContainer } = await render(
            afterTemplate,
            afterInput,
        );
        return diffOnlyChanges(container, afterContainer);
    };
}

function diffOnlyChanges(a, b) {
    const changes = diffLines(a, b);
    return changes
        .filter((part) => part.added || part.removed) // omit unchanged lines
        .map((part) => {
            const prefix = part.added ? "+ " : "- ";
            return part.value
                .split("\n")
                .filter((line) => line.trim() !== "") // clean up empty lines
                .map((line) => prefix + line)
                .join("\n");
        })
        .join("\n");
}
function diffHTML(beforeHTML) {
    const before = visualHTML(beforeHTML);
    return (afterHTML) => diffOnlyChanges(before, visualHTML(afterHTML));
}

export {
    snapshotHTML,
    diffServerSnapshots,
    diffHTML,
    diffOnlyChanges,
    visualHTML,
};
