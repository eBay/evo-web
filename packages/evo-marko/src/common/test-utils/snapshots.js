import { render } from "@marko/testing-library";
import { expect } from "vitest";

function snapshotHTML() {
  return async (template, input) => {
    const { container } = await render(template, input);
    expect(container).toMatchSnapshot();
  };
}

export { snapshotHTML };
