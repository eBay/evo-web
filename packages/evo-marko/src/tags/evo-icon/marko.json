{
  "taglib-id": "evo-marko-icons",
  "tags-dir": "./tags"
}
