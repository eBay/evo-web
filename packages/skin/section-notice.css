@import "./dist/section-notice/section-notice.css";
