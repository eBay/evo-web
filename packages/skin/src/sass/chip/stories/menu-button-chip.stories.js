export default { title: "Skin/Chip/Menu Button Chip" };

export const collapsed = () => `
<button class="chip chip--filter" aria-expanded="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const expanded = () => `
<button class="chip chip--filter" aria-expanded="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" aria-expanded="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-chevron-down-12"/>
        </svg>
    </button>
</div>
`;
