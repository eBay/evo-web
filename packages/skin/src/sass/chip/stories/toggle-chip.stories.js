export default { title: "Skin/Chip/Toggle Chip" };

export const button = () => `
<button class="chip" type="button" aria-pressed="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const buttonSelected = () => `
<button class="chip" type="button" aria-pressed="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const anchor = () => `
<a class="chip" herf="https://www.ebay.com">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--selected" herf="https://www.ebay.com">
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip" type="button" aria-pressed="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-close-12"></use>
        </svg>
    </button>
</div>
`;
