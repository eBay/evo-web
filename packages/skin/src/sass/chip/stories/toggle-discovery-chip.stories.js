export default { title: "Skin/Chip/Toggle Discovery Chip" };

export const button = () => `
<button class="chip chip--discovery" type="button" aria-pressed="false">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const buttonSelected = () => `
<button class="chip chip--discovery" type="button" aria-pressed="true">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const anchor = () => `
<a class="chip chip--discovery" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--discovery chip--selected" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" type="button" aria-pressed="false">
        <img
            class="chip__media"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
            alt=""
        >
        <span class="chip__text">Football</span>
    </button>
</div>
`;
