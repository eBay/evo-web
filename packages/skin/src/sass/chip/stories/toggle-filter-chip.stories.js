export default { title: "Skin/Chip/Toggle Filter Chip" };

export const button = () => `
<button class="chip chip--filter" type="button" aria-pressed="false">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>
    <span class="chip__text">Football</span>
</button>
`;

export const buttonSelected = () => `
<button class="chip chip--filter" type="button" aria-pressed="true">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football</span>
</button>
`;

export const anchor = () => `
<a class="chip chip--filter" herf="https://www.ebay.com">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football</span>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--filter chip--selected" herf="https://www.ebay.com">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--filter" type="button" aria-pressed="false">
        <svg class="icon icon--16" width="16" height="16">
            <use href="#icon-sneaker-16"/>
        </svg>
        <span class="chip__text">Football</span>
    </button>
</div>
`;
