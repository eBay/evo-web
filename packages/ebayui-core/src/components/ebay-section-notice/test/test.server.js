import { describe, it, expect } from "vitest";

import { render } from "@marko/testing-library";
import template from "../index.marko";
import * as mock from "./mock";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../section-notice.stories"; // import all stories from the stories file
const { Basic, WithCustomCTADismiss } = composeStories(stories);
const htmlSnap = snapshotHTML(__dirname);

describe("section-notice", () => {
    it("renders with status", async () => {
        const input = mock.SectionInfo;
        const { getByLabelText, getByText } = await render(template, input);
        const status = getByLabelText(input.a11yText).parentElement;
        expect(status).toMatchSnapshot();

        const containerUsingLabel = status.closest(
            `[aria-labelledby="${status.id}"]`,
        );
        expect(containerUsingLabel).toMatchSnapshot();

        const content = getByText(input.renderBody.text);
        expect(content).toMatchSnapshot();

        const container = content.parentElement;
        expect(container).toMatchSnapshot();
    });

    it("renders with light", async () => {
        const input = mock.SectionLight;
        const { getByText } = await render(template, input);
        const container = getByText(input.renderBody.text).parentElement;
        expect(container).toMatchSnapshot();

        const footer = getByText(input.footer.renderBody.text);
        expect(footer).toMatchSnapshot();
    });

    it("renders basic", async () => {
        await htmlSnap(Basic);
    });
    it("renders with custom CTA", async () => {
        await htmlSnap(WithCustomCTADismiss);
    });
});
