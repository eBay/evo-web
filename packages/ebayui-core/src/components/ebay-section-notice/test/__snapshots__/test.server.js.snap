// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`section-notice > renders basic 1`] = `
"[36m<DocumentFragment>[39m
  [36m<section[39m
    [33maria-labelledby[39m=[32m"s0-0-status"[39m
    [33maria-roledescription[39m=[32m"Notice"[39m
    [33mclass[39m=[32m"section-notice"[39m
    [33mrole[39m=[32m"region"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"section-notice__header"[39m
      [33mid[39m=[32m"s0-0-status"[39m
    [36m>[39m
      [36m<svg[39m
        [33maria-label[39m=[32m"attention"[39m
        [33mclass[39m=[32m"icon--attention-filled icon icon--16 icon--attention-filled"[39m
        [33mfocusable[39m=[32m"false"[39m
        [33mrole[39m=[32m"img"[39m
      [36m>[39m
        [36m<defs>[39m
          [36m<symbol[39m
            [33mid[39m=[32m"icon-attention-filled-16"[39m
            [33mviewBox[39m=[32m"0 0 16 16"[39m
          [36m>[39m
            [36m<path[39m
              [33md[39m=[32m"M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm0 12a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-4a1 1 0 1 1-2 0V5a1 1 0 0 1 2 0v3Z"[39m
            [36m/>[39m
          [36m</symbol>[39m
        [36m</defs>[39m
        [36m<use[39m
          [33mhref[39m=[32m"#icon-attention-filled-16"[39m
        [36m/>[39m
      [36m</svg>[39m
    [36m</div>[39m
    [36m<span[39m
      [33mclass[39m=[32m"section-notice__main"[39m
    [36m>[39m
      [36m<p>[39m
        [0mSection notice info. Things you need to know.[0m
      [36m</p>[39m
    [36m</span>[39m
  [36m</section>[39m
[36m</DocumentFragment>[39m"
`;

exports[`section-notice > renders with custom CTA 1`] = `
"[36m<DocumentFragment>[39m
  [36m<section[39m
    [33maria-labelledby[39m=[32m"s0-0-0-status"[39m
    [33maria-roledescription[39m=[32m"Notice"[39m
    [33mclass[39m=[32m"section-notice"[39m
    [33mrole[39m=[32m"region"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"section-notice__header"[39m
      [33mid[39m=[32m"s0-0-0-status"[39m
    [36m>[39m
      [36m<svg[39m
        [33maria-label[39m=[32m"information"[39m
        [33mclass[39m=[32m"icon--information-filled icon icon--16 icon--information-filled"[39m
        [33mfocusable[39m=[32m"false"[39m
        [33mrole[39m=[32m"img"[39m
      [36m>[39m
        [36m<defs>[39m
          [36m<symbol[39m
            [33mid[39m=[32m"icon-information-filled-16"[39m
            [33mviewBox[39m=[32m"0 0 16 16"[39m
          [36m>[39m
            [36m<path[39m
              [33md[39m=[32m"M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm1 11a1 1 0 1 1-2 0V8a1 1 0 0 1 2 0v3ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"[39m
            [36m/>[39m
          [36m</symbol>[39m
        [36m</defs>[39m
        [36m<use[39m
          [33mhref[39m=[32m"#icon-information-filled-16"[39m
        [36m/>[39m
      [36m</svg>[39m
    [36m</div>[39m
    [36m<span[39m
      [33mclass[39m=[32m"section-notice__main"[39m
    [36m>[39m
      [36m<h2[39m
        [33mclass[39m=[32m"section-notice__title"[39m
      [36m>[39m
        [36m<div>[39m
          [0mNotice title[0m
        [36m</div>[39m
      [36m</h2>[39m
      [36m<p>[39m
        [36m<strong>[39m
          [0mError:[0m
        [36m</strong>[39m
        [0m Please take another look at the following:[0m
      [36m</p>[39m
      [36m<p>[39m
        [36m<a[39m
          [33mhref[39m=[32m"#"[39m
        [36m>[39m
          [0mCard number[0m
        [36m</a>[39m
        [0m , [0m
        [36m<a[39m
          [33mhref[39m=[32m"#"[39m
        [36m>[39m
          [0mExpiration date[0m
        [36m</a>[39m
        [0m & [0m
        [36m<a[39m
          [33mhref[39m=[32m"#"[39m
        [36m>[39m
          [0mSecurity code[0m
        [36m</a>[39m
      [36m</p>[39m
    [36m</span>[39m
    [36m<p[39m
      [33mclass[39m=[32m"section-notice__cta"[39m
    [36m>[39m
      [36m<button[39m
        [33mclass[39m=[32m"fake-link"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [0mLink here[0m
      [36m</button>[39m
    [36m</p>[39m
    [36m<div[39m
      [33mclass[39m=[32m"section-notice__footer"[39m
    [36m>[39m
      [36m<button[39m
        [33maria-label[39m=[32m"Dismiss Notice"[39m
        [33mclass[39m=[32m"fake-link section-notice__dismiss"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--close-16 icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-close-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-close-16"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
    [36m</div>[39m
  [36m</section>[39m
[36m</DocumentFragment>[39m"
`;

exports[`section-notice > renders with light 1`] = `
<section
  aria-labelledby="s0-0-status"
  aria-roledescription="Notice"
  class="section-notice"
  role="region"
>
  <div
    class="section-notice__header"
    id="s0-0-status"
  />
  <span
    class="section-notice__main"
  >
    <!--F#13-->
    Content
    <!--F/-->
  </span>
  <div
    class="section-notice__footer"
  >
    <!--F#18-->
    Footer
    <!--F/-->
  </div>
</section>
`;

exports[`section-notice > renders with light 2`] = `
<div
  class="section-notice__footer"
>
  <!--F#18-->
  Footer
  <!--F/-->
</div>
`;

exports[`section-notice > renders with status 1`] = `
<div
  class="section-notice__header"
  id="s0-0-status"
>
  <svg
    aria-label="Heading Text"
    class="icon--information-filled icon icon--16 icon--information-filled"
    focusable="false"
    role="img"
  >
    <defs>
      <symbol
        id="icon-information-filled-16"
        viewBox="0 0 16 16"
      >
        <path
          d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm1 11a1 1 0 1 1-2 0V8a1 1 0 0 1 2 0v3ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
        />
      </symbol>
    </defs>
    <use
      href="#icon-information-filled-16"
    />
  </svg>
</div>
`;

exports[`section-notice > renders with status 2`] = `
<section
  aria-labelledby="s0-0-status"
  aria-roledescription="Notice"
  class="section-notice"
  role="region"
>
  <div
    class="section-notice__header"
    id="s0-0-status"
  >
    <svg
      aria-label="Heading Text"
      class="icon--information-filled icon icon--16 icon--information-filled"
      focusable="false"
      role="img"
    >
      <defs>
        <symbol
          id="icon-information-filled-16"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm1 11a1 1 0 1 1-2 0V8a1 1 0 0 1 2 0v3ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-information-filled-16"
      />
    </svg>
  </div>
  <span
    class="section-notice__main"
  >
    <!--F#13-->
    Content
    <!--F/-->
  </span>
</section>
`;

exports[`section-notice > renders with status 3`] = `
<span
  class="section-notice__main"
>
  <!--F#13-->
  Content
  <!--F/-->
</span>
`;

exports[`section-notice > renders with status 4`] = `
<section
  aria-labelledby="s0-0-status"
  aria-roledescription="Notice"
  class="section-notice"
  role="region"
>
  <div
    class="section-notice__header"
    id="s0-0-status"
  >
    <svg
      aria-label="Heading Text"
      class="icon--information-filled icon icon--16 icon--information-filled"
      focusable="false"
      role="img"
    >
      <defs>
        <symbol
          id="icon-information-filled-16"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm1 11a1 1 0 1 1-2 0V8a1 1 0 0 1 2 0v3ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-information-filled-16"
      />
    </svg>
  </div>
  <span
    class="section-notice__main"
  >
    <!--F#13-->
    Content
    <!--F/-->
  </span>
</section>
`;
